pkg/scheduler/scheduler.go

```diff
@@ -745,14 +745,14 @@ func (s *Scheduler) establishEstimatorConnections() {
 }
 
 // patchBindingStatusCondition patches schedule status condition of ResourceBinding when necessary.
-func patchBindingStatusCondition(karmadaClient karmadaclientset.Interface, rb *workv1alpha2.ResourceBinding, newScheduledCondition metav1.Condition) error {
+func patchBindingStatusCondition(karmadaClient karmadaclientset.Interface, rb *workv1alpha2.ResourceBinding, newScheduledCondition metav1.Condition, ignoreErr bool) error {
 	klog.V(4).Infof("Begin to patch status condition to ResourceBinding(%s/%s)", rb.Namespace, rb.Name)
 
 	updateRB := rb.DeepCopy()
 	meta.SetStatusCondition(&updateRB.Status.Conditions, newScheduledCondition)
-	// Postpone setting observed generation until schedule succeed, assume scheduler will retry and
-	// will succeed eventually.
-	if newScheduledCondition.Status == metav1.ConditionTrue {
+	// If an error during scheduling could be ignored, the scheduling result would have been patched successfully.
+	// The scheduler observed generation should be updated.
+	if ignoreErr {
```

Comment:
I feel that this judgment condition `newScheduledCondition.Status == metav1.ConditionTrue` also needs to be retained, the relationship between the two is OR. What do you think?

Reply:
https://github.com/ctripcloud/karmada/blob/48b8e9f9ba8a4e09a58524631aa596e84b8a8fb4/pkg/scheduler/helper.go#L95  

In fact, after `getConditionByError`, `ignoreErr` would be `true` when condition is `true`, so I only judge `ignoreErr` here.  

But it seems to be easily misunderstanding, I will fix it soon.

Reply:
Done.